legacy/reqcounter/reqcounter.go

```diff
@@ -16,30 +16,73 @@ limitations under the License.
 
 package reqcounter
 
-import "sync"
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
 
 // RequestCounter records the number of HTTP requests to GCR.
 type RequestCounter struct {
-	mutex    sync.Mutex
-	requests uint64
+	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
+	requests uint64     // Number of HTTP requests since recording started.
+	quota    uint64     // Maximum requests allowed before GCR throttling occurs.
+	since    time.Time  // When the current counter began recording requests.
+}
+
+// log records the number of HTTP requests found and resets the counter.
+func (rc *RequestCounter) log() {
+	// Grab the lock to prevent race-conditions.
+	rc.mutex.Lock()
+	defer rc.mutex.Unlock()
+	// Log the number of requests found.
+	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", counter.since.Format("01-02-2006 15:04:05"), counter.requests)
+	logrus.Debug(msg)
+	// Reset the counter.
+	rc.reset()
```

Comment:
To make it easier to search our logs from the GCP console, can you have 2 counters?

1. a global counter that always increments and is never reset
2. a counter that is reset each measurement window (you already have this)

And, can you log the 1st counter as well whenever we log the 2nd counter? Just seeing a running total will be useful.

Reply:
Thinking about this a bit more, it would be helpful to expand counters (if necessary) to log how many requests were made in the last:

1. 10 minutes (we already do this in this PR)
2. 1 hour (not yet implemented)
3. 24-hour period (not yet implemented)

The request counter information for the previous 1-hour and 24-hour periods would both be a rolling on a 10-minute basis. That is, every 10 minutes the log would say:

```
There were X HTTP requests to GCR made in the last 10 minutes.
There were Y HTTP requests to GCR made in the last hour.
There were Z HTTP requests to GCR made in the last 24 hours.
```

Reply:
Could this be added in a following PR?

Reply:
Added an extra TODO within #358  "Create multiple counters that log at different offsets"